### PR TITLE
Fixes add network policy locking

### PR DESCRIPTION
**Pull request by trozet**
Ensure the policy is set correctly with proper locking before we start
the watchers to update the policy attributes.

Signed-off-by: Tim Rozet <trozet@redhat.com>

### CODE REVIEW

**Comment by trozet** [comment]
@dcbw @danwinship @girishmg PTAL

**Comment by trozet** [comment]
@dave-tucker  PTAL

File: go-controller/pkg/ovn/policy.go [hunk 1]
```diff
@@ -959,6 +959,14 @@ func (oc *Controller) addNetworkPolicy(policy *knet.NetworkPolicy) {
 		return
 	}
 
+	type policyHandler struct {
+		gress             *gressPolicy
+		namespaceSelector *metav1.LabelSelector
+		podSelector       *metav1.LabelSelector
+		addrSet           string
+		peerMap           map[string]bool
+	}
```
Comment (danwinship) [inline go-controller/pkg/ovn/policy.go:968]: this could maybe just all be added to `gressPolicy`?

Comment (trozet) [reply go-controller/pkg/ovn/policy.go:968]: @danwinship yeah there are a bunch of improvements (like the one you mentioned about watchers yesterday) to this code and it is going to be refactored anyway. I don't want to make changes other than what is needed to just fix this locking bug in this commit.

File: go-controller/pkg/ovn/policy.go [hunk 2]
```diff
@@ -1057,41 +1053,43 @@ func (oc *Controller) addNetworkPolicy(policy *knet.NetworkPolicy) {
...
+				podSelector:       toJSON.PodSelector,
+				addrSet:           hashedLocalAddressSet,
+				peerMap:           peerPodAddressMap,
 			}
+			policyHandlers = append(policyHandlers, egressPolicyHandler)
 		}
 		np.egressPolicies = append(np.egressPolicies, egress)
 	}
-
+	np.Unlock()
```
Comment (dcbw) [inline go-controller/pkg/ovn/policy.go:1067]: @trozet why unlock here versus right after the handleLocalPodSelector()?

Comment (dcbw) [reply go-controller/pkg/ovn/policy.go:1067]: nevermind

**Pull request merged**